Dashboard/app/js/lib/views/maps/map-views-common.js

```diff
@@ -887,80 +335,21 @@ FLOW.NavMapsView = FLOW.View.extend({
     this.polygons.push(geoShape);
   },
 
-  checkHierarchy: function(parentFolderId){
-    var self = this;
-
-    //if survey hierarchy object has previously been retrieved, no need to pull it anew
-    if(self.hierarchyObject.length > 0){
-      self.manageHierarchy(parentFolderId);
-    }else{
-      $.get(
-        '/rest/survey_groups', /*place survey_groups endpoint here*/
-        function(data, status){
-          if(data['survey_groups'].length > 0){
-            self.hierarchyObject = data['survey_groups'];
-            self.manageHierarchy(parentFolderId);
-          }
-        });
-    }
-  },
-
-  manageHierarchy: function(parentFolderId){
-    var self = this;
-
-    rows = self.hierarchyObject;
-    rows.sort(function(el1, el2) {
-      return self.compare(el1, el2, 'name');
-    });
-
-    //create folder and/or survey select element
-    var folder_survey_selector = $("<select></select>").attr("class", "folder_survey_selector");
-    folder_survey_selector.append('<option value="">--' + Ember.String.loc('_choose_folder_or_survey') + '--</option>');
-
-    for (var i=0; i<rows.length; i++) {
-      //append return survey list to the survey selector element
-      var surveyGroup = rows[i];
-
-      //if a subfolder, only load folders and surveys from parent folder
-      if(surveyGroup.parentId == parentFolderId){
-        folder_survey_selector.append('<option value="'
-          + surveyGroup.keyId + '"'
-          +'data-type="'+surveyGroup.projectType+'">'
-          + surveyGroup.name
-          + '</option>');
-      }
-    }
-    $("#survey_hierarchy").append(folder_survey_selector);
-  },
-
-  cleanHierarchy: function(element){
-    $(element).nextAll().remove();
-  },
-
-  clearCartodbLayer: function(){
-    //check to confirm that there are no layers displayed on the map
-    if(this.layerExistsCheck){
-      this.map.removeLayer(this.cartodbLayer);
-      this.layerExistsCheck = false;
-    }
-  },
-
   clearMap: function() {
     var self = this;
-    if(self.marker != null){
+    self.set('detailsPaneVisible', false);
+    if (self.marker) {
       self.map.removeLayer(self.marker);
-      self.hideDetailsPane();
-      $('#pointDetails').html('<p class="noDetails">'+Ember.String.loc('_no_details') +'</p>');
     }
 
-    if(!$.isEmptyObject(self.mediaMarkers)) {
-      for(mediaMarker in self.mediaMarkers) {
+    if (!$.isEmptyObject(self.mediaMarkers)) {
```

Comment:
In the same spirit of trying to use our framework of choice I would suggest using `Ember.empty(...)`